### PR TITLE
[第2組]Refactor/naming conventions

### PR DESCRIPTION
**Pull request by Yuannzh1**
### Purpose
Refactor naming conventions for better readability.

### Changes
Renamed all functions, variables.

### CODE REVIEW

**Comment by Steven0811** [comment]
Some variables still have not been changed to be more readable names, such as "n" in line 367 and "u" in line 484. Magic numbers are not extracted, too.

**Comment by WANG-YongLE** [comment]
優點：
Commit 內容清楚，每個 commit 只包含單一更動。
建議在 PR 說明中提供更詳細的描述。
建議：
77行 bgs建議改成scheduler
218行 p 建議改成paragraphinfo
276行 p1 改成password，p2改成hashedpassword
340行 uid改成userid
367行 n 改成 newscontent
484行 u改成user
490行 n_id u_id改成newsid和userid

**Comment by kevin0216** [comment]
每個 Commit 僅包含一項改動，各 Commit 的更動內容皆明確，但可能會造成 Commit 過長，可能需要進行 Commit Squash。另外以下是幾點建議:
- Line 480 後較多沒修改到的地方，如 id (此名稱會與內部方法衝突) / u.id 等
- LLM Prompt 如 Line 195 的部分會建議用更明確的名稱表達，如 ai_prompt / prompt 等，message 可能會較不精確
- Line 337 的 _id_counter 前的底線是指 private variable，建議不要修改 (因現階段還不確定是否為 Private)

**Comment by Yi-Cheng-Wang** [comment]
**Evaluation:**
- Degree of variable refactoring is slightly low
- Variable renaming is decent
- Code layout is tidy
- Although there are many commits, they are very meaningful
- Clarity of the PR is low

**Suggestions:**
- Magic numbers seem to be left unaddressed; for example, the value `1892dhianiandowqd0n` on line 290 could be passed in a better way.
- The variable `ai` on line 202 could have a more meaningful name.
- The variable `bgs` on line 259 is not easily understandable and could be improved.
- The parameters passed to the function on lines 486 and 490 have not been changed, and `u` and `n` are not good variable names.

**Comment by mlcp29th-hi** [comment]
Hello,
* 建議命名時若有更好的選擇，最好盡量避免諸如 data/info/result 等形式的命名，否則以旁人的角度較不容易立刻理解 `news`、`new_data`、`news_info`、`aggregated_news_data` 等的差別，畢竟變數本身即資料，`_data` 等後綴對於理解其功能的幫助有限。

* 根據 [PEP 8](https://peps.python.org/pep-0008/)，以單個底線命名的成員有特殊目的，建議維持原狀：
 > _single_leading_underscore: weak “internal use” indicator. E.g. from M import * does not import objects whose names start with an underscore.
* `get_news_article()` 會使人預期呼叫此函式可得到 NewArticle 物件，但此函式僅由 background_scheduler 定期觸發執行下載資料、存入資料庫而沒有回傳值，且實際讀取新聞資料的程式也並非使用此函式，或許可以考慮另做命名。

* `verify_hashed_password` 的功能是「驗證輸入的 password 經過 hash 後和 user.hashed_password」是否一致，或許以 `verify_password` 命名會更符合其目的，此外，此函式會回傳 boolean 表示正確與否，這樣的函式命名時經常以 `is_`、`has_`、`does_` 開頭，在語意上會更流暢如 `is_password_correct()`。

* `get_news_exists_status` 建議能改成 `get_news_existence_status` 或是如同上述原因能考慮改成 `does_news_exist`。

* 基於一個 Pull Request 應該只專注於一個目的的原則，docker-compose.yml  因加入 sonarqube 所做的更動放在其他的 Pull Request 會更好。

File: backend/main.py [hunk 1]
```diff
@@ -375,9 +375,9 @@ def read_news(db=Depends(session_opener)):
 @app.get(
     "/api/v1/news/user_news"
 )
-def read_user_news(
+def get_user_upvoted_news(
```
Comment (taro0513) [inline backend/main.py:378]: This is not the function to get the news that specific users have upvoted. The purpose of the function, according to the code we get, is that we get all news in the database "with" the value that if the user has upvoted the news

**Review by taro0513 — COMMENTED** [review]
The commit messages are very detailed that we can easily find out the change in one specific commit : )

**Pull request merged**